collect_app/src/androidTest/java/org/odk/collect/android/support/pages/NotificationDrawer.kt

```diff
@@ -31,36 +32,30 @@ class NotificationDrawer {
     fun assertNotification(
         appName: String,
         title: String,
-        subtext: String,
+        subtext: String? = null,
         body: String? = null
     ): NotificationDrawer {
         val device = waitForNotification(appName)
 
         val titleElement = device.findObject(By.text(title))
         assertThat(titleElement, not(nullValue()))
 
-        var subtextElement = device.findObject(By.text(subtext))
-        if (subtextElement == null) {
-            device.findObject(By.text(appName)).click() // Expand notification to show subtext
-            subtextElement = device.findObject(By.text(subtext))
+        if (subtext != null) {
+            assertExpandedText(device, appName, subtext)
         }
 
-        assertThat(subtextElement.text, `is`(subtext))
-
-        body?.let {
-            val bodyElement = device.findObject(By.text(body))
-            assertThat(bodyElement, not(nullValue()))
+        if (body != null) {
+            assertExpandedText(device, appName, body)
         }
 
-        assertExpandedText(device, appName, subtext)
-
         return this
     }
 
     fun <D : Page<D>> clickAction(
```

Comment:
This action seems to always receive the same set of values so maybe it would be good to add them as defaults.

Reply:
The values are always passed from other methods though, so it would mean adding a default app name to `NotificationDrawer`. I'd rather avoid that as there is currently no reason for `NotificationDrawer` to know anything about Collect.